src/hotspot/share/utilities/ostream.cpp

```diff
@@ -961,13 +975,12 @@ void ostream_exit() {
   if (ostream_exit_called)  return;
   ostream_exit_called = true;
   ClassListWriter::delete_classlist();
-  if (tty != defaultStream::instance) {
+  // Make sure tty works after VM exit by assigning an always-on functioning fdStream.
+  if (tty != &tty_preinit_stream && tty != defaultStream::instance) {
     delete tty;
   }
-  if (defaultStream::instance != NULL) {
-    delete defaultStream::instance;
-  }
-  tty = NULL;
+  tty = DisplayVMOutputToStderr ? fdStream::stdout_stream() : fdStream::stderr_stream();
```

Comment:
Shouldn't this be something like:
```
tmp = tty;
tty = ...
delete tmp;
```

Reply:
It probably does not make much difference since the swap-over is not atomic anyway, but I can change it.